python/cudf_polars/cudf_polars/dsl/ir.py

```diff
@@ -454,20 +514,20 @@ def evaluate(self, *, cache: MutableMapping[int, DataFrame]) -> DataFrame:
             df = DataFrame.from_table(
                 plc_tbl_w_meta.tbl, plc_tbl_w_meta.column_names(include_children=False)
             )
-            col_order = list(self.schema.keys())
+            col_order = list(schema.keys())
             # TODO: remove condition when dropping support for polars 1.0
             # https://github.com/pola-rs/polars/pull/17363
```

Comment:
Can this condition be dropped now? Our polars minimum is higher than 1.0.

Reply:
There is an `if` directly below this comment that was modified in this PR. I am guessing that is the logic that could be removed.

Reply:
We need the second part of the condition, but we can drop the `row_index is not None` part

Reply:
>we can drop the row_index is not None part, and then remove these two comment lines.

We check `if row_index is not None` in several places within `do_evaluate`. Do we only care about this specific instance below this comment, or is the `row_index: tuple[str, int] | None` typing in the function signature also a workaround for the same bug?

The `if col_order is not None:` check below also seems unnecessary, since the `col_order = list(schema.keys())` definition above suggests we will always enter this block anyway.

Reply:
Ah sorry, I got it wrong. Maybe this?
```
diff --git a/python/cudf_polars/cudf_polars/dsl/ir.py b/python/cudf_polars/cudf_polars/dsl/ir.py
index 09ef893489..69df2a6f91 100644
--- a/python/cudf_polars/cudf_polars/dsl/ir.py
+++ b/python/cudf_polars/cudf_polars/dsl/ir.py
@@ -456,12 +456,9 @@ class Scan(IR):
                 plc_tbl_w_meta.tbl, plc_tbl_w_meta.column_names(include_children=False)
             )
             col_order = list(self.schema.keys())
-            # TODO: remove condition when dropping support for polars 1.0
-            # https://github.com/pola-rs/polars/pull/17363
-            if row_index is not None and row_index[0] in self.schema:
+            if row_index is not None:
                 col_order.remove(row_index[0])
-            if col_order is not None:
-                df = df.select(col_order)
+            df = df.select(col_order)
         else:
             raise NotImplementedError(
                 f"Unhandled scan type: {self.typ}"
```

Reply:
Okay, thanks! Changed.